guardian/mobile-save-for-later
# Add cdk-defined version of API infrastructre

**N.B. this PR depends on https://github.com/guardian/mobile-save-for-later/pull/64**

## What does this change?

We are migrating the `mobile-save-for-later` service from CloudFormation to `@guardian/cdk`. The migration will involve 5 PRs:

1. Add `@guardian/cdk` to repository; update CI & deployment wiring accordingly (https://github.com/guardian/mobile-save-for-later/pull/64)
**2. Provision a new version of the infrastructure using `@guardian/cdk` [this PR]**
3. Move DNS configuration from CloudFormation into `@guardian/cdk` (but continue routing requests to old infrastructure)
4. Update DNS configuration (start routing requests to new infrastructure)
5. Remove old infrastructure

## How to test

I've deployed this to `CODE` to confirm that the deployment works as expected.

I also temporarily updated DNS so that requests were being routed to this new infrastructure. Then I used a debug version of the app to confirm that saving articles/viewing saved articles still works as expected. (I checked the AWS Lambda logs and DynamoDB to confirm that the new Lambdas were being exercised as expected).

## How can we measure success?

We are a step closer to defining all infrastructure using `@guardian/cdk`! 

## Have we considered potential risks?

Yes, the risk is very low at this stage. Although we have provisioned a new API/Lambdas, the old API/Lambdas still serve all of the traffic.
